src/Rings/mpoly-ideals.jl

```diff
@@ -1890,13 +1890,16 @@ julia> dim(I)
 1
 ```
 """
-@attr Int function dim(I::MPolyIdeal)
-  if I.dim > -1
-    return I.dim
+function dim(I::MPolyIdeal)
+  if I.dim === nothing
+    if is_zero(ngens(base_ring(I))) # Catch a boundary case
+      I.dim = 0
```

Comment:
Hm, is this correct if the coefficient ring is not a field?

Reply:
I guess you're right. It was probably wrong before, but we should correct it.